servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancerBuilder.java

```diff
@@ -94,19 +102,22 @@ static final class DefaultLoadBalancerFactory<ResolvedAddress, C extends LoadBal
 
         private final String id;
         private final LoadBalancingPolicy<ResolvedAddress, C> loadBalancingPolicy;
+        private final int subsetSize;
         @Nullable
         private final LoadBalancerObserverFactory loadBalancerObserverFactory;
         private final ConnectionPoolStrategyFactory<C> connectionPoolStrategyFactory;
         private final OutlierDetectorConfig outlierDetectorConfig;
         private final Executor executor;
 
         DefaultLoadBalancerFactory(final String id, final LoadBalancingPolicy<ResolvedAddress, C> loadBalancingPolicy,
+                                   final int subsetSize,
                                    @Nullable final LoadBalancerObserverFactory loadBalancerObserverFactory,
                                    final ConnectionPoolStrategyFactory<C> connectionPoolStrategyFactory,
                                    final OutlierDetectorConfig outlierDetectorConfig,
                                    final Executor executor) {
             this.id = requireNonNull(id, "id");
             this.loadBalancingPolicy = requireNonNull(loadBalancingPolicy, "loadBalancingPolicy");
+            this.subsetSize = ensurePositive(subsetSize, "subsetSize");
```

Comment:
optional: we only need to validate user input in the builder's setter, and inside internal constructors we can trust that the value is already validated

Reply:
Do you want me to remove all the validation, including the null checks? I kept it here only because it was the pattern established by RR LB, but that alone doesn't mean we need to continue it as it does seem like an outlier.

Reply:
Taking into account that this happens once per new client, it won't hurt to have validation in multiple places. So, up to you. We can either leave it as-is or avoid extra validation for subset size in this PR and clean up other checks in a follow up

Reply:
I'll leave it as is and we can make another PR to remove excess validation later if we feel inclined.